docs/batch_inference_with_mms.md

```diff
@@ -0,0 +1,368 @@
+    1. Logic for `postprocess`ing: The handler should take the output of the inference-logic and `postprocess` this data. This is then sent back the clients as the final result.
+
+Let's look into how we define the `initialize`, `preprocess`, `inference` and `postprocess` logic in detail.
+
+#### Initialization logic
+As a part of the handler's parameters, MMS sends a `context` object to the handler. Besides everything else, this `context` object also contains the batch-size that this model was configured to handle. 
+The configuration for the model comes from the `POST /models` call and this is explained in detail in the below section [MMS Model Configuration](#mms-model-configuration). Lets look at a code snippet,
+
+```python
+class MXNetVisionServiceBatching(object):
+...
+    def initialize(self, context):
+        ...
+        self._batch_size = context.system_properties["batch_size"]
+        data_shapes = []
+        # Read the data shape
+        data_shapes[0] = self._batch_size
+        ...
+        # Bind the MXNet module with the data shape NCHW where N is the batch size
+        self.mx_model.bind(for_training=False, data_shapes=data_shapes)
+        ...
+...        
+```
+This initialization logic binds the MXNet module with a batch-size that the model is configured with. For example, if the incoming requests are images with RGB layers and the input to the network is
+or size [3,224, 224], the batch adds an additional dimension to this data. If the batch size configured with `POST /models` API is 8, the input to the DL Framework becomes [8, 3, 224, 224]. 
+
+#### Preprocess logic
+Once a model is initialized, the DL framework (in our case MXNet) expects a `batch_size` number of requests. Hence, the preprocessing logic should make sure that no matter how many requests
+come into the handler, it always returns the `batch_size` number of NDArray elements to be used in the inference logic. Lets look at a sample `preprocess` logic below 
+
+```python
+import mxnet as mx
+
+class MXNetVisionServiceBatching(object):
+...
+    def initialize(self, context):
+    ...
+...
+    def preprocess(self, request):
+        img_list = []
+        param_name = self.signature['inputs'][0]['data_name']
+        input_shape = self.signature['inputs'][0]['data_shape']
+        # We are assuming input shape is NCHW
+        [c, h, w] = input_shape[1:]
+
+        for idx, data in enumerate(request):
+            img = data.get(param_name)
+            if img is None:
+                img = data.get("body")
+
+            if img is None:
+                img = data.get("data")
+
+            if img is None or len(img) == 0:
+                logging.error("Error processing request")
+                self.erroneous_reqs.add(idx)
+                continue
+
+            try:
+                img_arr = mx.image.imdecode(img, 1, True, None)
+            except Exception as e:
+                logging.warn(e, exc_info=True)
```

Comment:
logging.warn is deprecated

Reply:
True. I ignored this warning. I can make this an error.